airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/state/StreamsManager.kt

```diff
@@ -185,6 +185,9 @@ class DefaultStreamManager(
     }
 }
 
+// The purpose of a @Factory is to instantiate beans for interfaces or classes we don't control
+// and which were defined in libraries which didn't care to annotate them in a useful way.
+// This is not the case here.
```

Comment:
Hm... I thought I was backed into this by Micronaut not injecting non-val constructor parameters, but I tested and that's not the case. Agree that this doesn't need to be a factory.